client/src/grpc_client.rs

```diff
@@ -25,32 +25,51 @@ use libra_types::{
     },
     transaction::{Transaction, Version},
     vm_error::StatusCode,
+    waypoint::Waypoint,
 };
 use std::convert::TryFrom;
 use std::sync::{Arc, Mutex};
 
 const MAX_GRPC_RETRY_COUNT: u64 = 1;
 
+struct TrustedState {
+    version: Version,
+    epoch_info: EpochInfo,
+    latest_epoch_change_li: Option<LedgerInfoWithSignatures>,
+}
+
 /// Struct holding dependencies of client, known_version_and_epoch is updated when learning about
 /// new LedgerInfo
 pub struct GRPCClient {
     client: AdmissionControlClient,
-    known_version_and_epoch: Arc<Mutex<(Version, EpochInfo)>>,
+    state: Arc<Mutex<TrustedState>>,
+    waypoint: Option<Waypoint>,
```

Comment:
imo this is better to be an enum
```
enum KnownState {
  Waypoint(Waypoint),
  VerifiedState(VersionAndEpoch),
}
```
we'll start with a waypoint and switch to the TrustedState once verification passed